### PR TITLE
AC-2478::Search results summary information not announced to screen r…

### PR DESCRIPTION
**Pull request by glo82145**
## Description

[NODE][body>div:nth-of-type(1)>*:nth-child(1)>*:nth-child(2)]

1. With JAWS operating, press Tab to move through the content.
2. When keyboard focus is on Search, press Enter or Space to activate it.
3. Enter value to search on and press enter.

Actual Behavior
Screen reader users are not notified when "No results were found" or number of results returned for search.

When developers don't specify which portions of a page are live regions, screen readers and other assistive technologies will not know which areas of the page to monitor for updates.

Expected Behavior
Ensure screen readers are notified of summary of the number of search results returned using aria-live="polite".

## Related Issue

Closes https://jira.corp.magento.com/browse/AC-2478

## Verification Steps

✔️ QA Passed
Pre-Conditions:

1. Have Magento instance with sample data installed

2. Make sure to have pwa studio installed

3. Make sure to have a customer login for front end login

 

Manual Steps executed:

Enable screen reader extension
navigate to serach button and serach any product 
observe the response by screen reader when No results were found for the serached product 
observe the response by screen reader when number of results returned for searched product{}
 

✖️ Behaviour Before The Fix : 

Screen reader users are not notified when "No results were found" or number of results returned for search.

✔️Behaviour After The Fix: 

Screen reader users are  notified when "No results were found" or number of results returned for search.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### Resolved issues:
1. [x] resolves magento/pwa-studio#3822: AC-2478::Search results summary information not announced to screen r…

### CODE REVIEW

**Comment by supernova-at** [comment]
@magento create issue from PR

**Review by michaelyu0123 — CHANGES_REQUESTED** [review]

File: packages/venia-ui/lib/components/SearchPage/searchPage.js [hunk 1]
```diff
@@ -168,8 +172,8 @@ const SearchPage = props => {
     );
 
     const itemCountHeading =
-        data && !loading ? (
-            <span className={classes.totalPages}>
+        data && !loading && data.products.items.length > 0 ? (
```
[the remark this answers is not in the document]
Comment (glo82145) [reply packages/venia-ui/lib/components/SearchPage/searchPage.js:175]: Hi @michaelyu0123 .. I have just added a condition  extra and no information has been removed from here. Could please explain more for better understanding

[the remark this answers is not in the document]
Comment (supernova-at) [reply packages/venia-ui/lib/components/SearchPage/searchPage.js:175]: Hi @glo82145 , good question - take a look at the changes in the snapshot file above.

```
        <span
          className="totalPages"
        >
          0 items
        </span>
```

This `span` no longer gets rendered when there aren't any search results (`data.products.items.length = 0`). @michaelyu0123 is pointing out that we still do want to show the user the information in this case.

**Review by michaelyu0123 — CHANGES_REQUESTED** [review]

File: packages/venia-ui/lib/components/SearchPage/searchPage.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nishant04412) [reply packages/venia-ui/lib/components/SearchPage/searchPage.js]: @michaelyu0123 @supernova-at We have made the required changes. Now the behaviour is like '0 items' is showing on UX and screen reader is announcing 'No results found'. Please check and update if the code changes are acceptable.

**Review by michaelyu0123 — DISMISSED** [review]

**Comment by dpatil-magento** [comment]
run cypress

**Comment by dpatil-magento** [comment]
run cypress @checkout

**Comment by dpatil-magento** [comment]
run cypress @checkout

**Comment by dpatil-magento** [comment]
run lighthouse-mobile

**Comment by dpatil-magento** [comment]
run lighthouse-desktop

**Review by dpatil-magento — APPROVED** [review]
QA Approved.

**Pull request merged**